The job **`x86_64-apple-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/8872324230/job/24356511147) [(plain)](https://github.com/rust-lang-ci/rust/commit/c016833569e9eddb7d5c29a7acb9581b95177a04/checks/24356511147/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
failures:

---- [assembly] tests/assembly/manual-eq-efficient.rs stdout ----

error: verification with 'FileCheck' failed
status: exit status: 1
command: "/Users/runner/work/rust/rust/build/x86_64-apple-darwin/ci-llvm/bin/FileCheck" "--input-file" "/Users/runner/work/rust/rust/build/x86_64-apple-darwin/test/assembly/manual-eq-efficient/manual-eq-efficient.s" "/Users/runner/work/rust/rust/tests/assembly/manual-eq-efficient.rs" "--check-prefix=CHECK" "--check-prefix" "NONMSVC" "--allow-unused-prefixes" "--dump-input-context" "100"
--- stderr -------------------------------
--- stderr -------------------------------
/Users/runner/work/rust/rust/tests/assembly/manual-eq-efficient.rs:20:17: error: CHECK-NEXT: is not on the line after the previous match
 // CHECK-NEXT: ret
                ^
/Users/runner/work/rust/rust/build/x86_64-apple-darwin/test/assembly/manual-eq-efficient/manual-eq-efficient.s:17:2: note: 'next' match was here
 ^
 ^
/Users/runner/work/rust/rust/build/x86_64-apple-darwin/test/assembly/manual-eq-efficient/manual-eq-efficient.s:15:9: note: previous match ended here
 sete al
        ^
/Users/runner/work/rust/rust/build/x86_64-apple-darwin/test/assembly/manual-eq-efficient/manual-eq-efficient.s:16:1: note: non-matching line after previous match is here
 pop rbp

Input file: /Users/runner/work/rust/rust/build/x86_64-apple-darwin/test/assembly/manual-eq-efficient/manual-eq-efficient.s
Check file: /Users/runner/work/rust/rust/tests/assembly/manual-eq-efficient.rs


-dump-input=help explains the following input dump.
Input was:
<<<<<<
<<<<<<
         1:  .section __TEXT,__text,regular,pure_instructions 
         2:  .macosx_version_min 10, 12 
         3:  .intel_syntax noprefix 
         4:  .globl _manual_eq 
         5:  .p2align 4, 0x90 
         6: _manual_eq: 
         7:  .cfi_startproc 
         8:  push rbp 
         9:  .cfi_def_cfa_offset 16 
        10:  .cfi_offset rbp, -16 
        11:  mov rbp, rsp 
        12:  .cfi_def_cfa_register rbp 
        13:  mov eax, dword ptr [rdi] 
        14:  cmp eax, dword ptr [rsi] 
        15:  sete al 
        16:  pop rbp 
        17:  ret 
next:20      !~~  error: match on wrong line
        18:  .cfi_endproc 
        20: .subsections_via_symbols 
>>>>>>
------------------------------------------

```

</details>

--- a/tests/assembly/manual-eq-efficient.rs
+++ b/tests/assembly/manual-eq-efficient.rs
@@ -1,0 +1,22 @@
+// Regression test for #106269
+//@ assembly-output: emit-asm
+//@ compile-flags: --crate-type=lib -O -C llvm-args=-x86-asm-syntax=intel
+//@ only-x86_64
+//@ ignore-sgx
+
+pub struct S {
+    a: u8,
+    b: u8,
+    c: u8,
+    d: u8,
+}
+
+// CHECK-LABEL: manual_eq:
+#[no_mangle]
+pub fn manual_eq(s1: &S, s2: &S) -> bool {
+    // CHECK: mov [[REG:[a-z0-9]+]], dword ptr [{{[a-z0-9]+}}]
+    // CHECK-NEXT: cmp [[REG]], dword ptr [{{[a-z0-9]+}}]
+    // CHECK-NEXT: sete al
+    // CHECK: ret
+    s1.a == s2.a && s1.b == s2.b && s1.c == s2.c && s1.d == s2.d
+}

